Analysis/Tasks/PWGHF/HFD0ToPiKCandidateSelector.cxx

```diff
@@ -161,39 +210,93 @@ struct HFD0CandidateSelector {
   bool selectionTopolConjugate(const T1& hfCandProng2, const T2& trackPion, const T2& trackKaon)
   {
 
+    bool selectionDebug = true;
+    bool checkD0 = false;
+    if (trackPion.charge() > 0) {
+      checkD0 = true;
+    }
     auto candpT = hfCandProng2.pt();
     int pTBin = getpTBin(candpT);
     if (pTBin == -1) {
       return false;
     }
 
-    if (trackPion.charge() > 0) { //invariant mass cut
+    if (checkD0) { //invariant mass cut
       if (TMath::Abs(InvMassD0(hfCandProng2) - RecoDecay::getMassPDG(421)) > cuts[pTBin][0]) {
-        return false;
+        if (!b_Debug) {
+          return false;
+        } else {
+          selectionTopolConjugateBit |= 1 << 0;
+          selectionDebug = false;
+        }
       }
     } else {
       if (TMath::Abs(InvMassD0bar(hfCandProng2) - RecoDecay::getMassPDG(421)) > cuts[pTBin][0]) {
-        return false;
+        if (!b_Debug) {
+          return false;
+        } else {
+          selectionTopolConjugateBit |= 1 << 1;
+          selectionDebug = false;
+        }
       }
     }
 
     if (trackPion.pt() < cuts[pTBin][4] || trackKaon.pt() < cuts[pTBin][3]) {
-      return false; //cut on daughter pT
+      if (!b_Debug) {
+        return false;
+      } else {
+        if (checkD0) {
+          selectionTopolConjugateBit |= 1 << 2;
+        } else {
+          selectionTopolConjugateBit |= 1 << 3;
+        }
+        selectionDebug = false;
+      }
```

Comment:
This can be simplified to
```c++
      if (!b_Debug) {
        return false;
      }
      selectionTopolConjugateBit |= 1 << (checkD0 ? 2 : 3);
      selectionDebug = false;
```